app/controllers/assistants/instructions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Assistants::InstructionsController < ApplicationController
+  before_action :set_assistant
+
```

Comment:
@robacarp what’s the rationale behind a dedicated controller just for the instructions? I would have assumed we could reuse the assistants_controller#update.

For example, the “create GPT” page in ChatGPT lets you configure many aspects of an Assistant all at once (name, image, instructions, etc). We’d surely want that to just be a single post to a single endpoint to save all those attributes. So even though this new view is only letting you update this single aspect of a GPT, we can still use the same general #update controller.

Reply:
@krschacht I considered that. I broke out the #edit action because it has a dedicated view associated, with it's own form. That could also be handled within the Assistants controller (#edit_instructions and edit_instructions.html.erb), but my experience has been that staying pedantically true to one-controller-one-idea is better in the long run. [This blog post from 2016](http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/) comes to mind.

So I settled onto this approach because, as we have it in the product, the idea of updating an assistant's instructions is different from the idea of updating the assistant's other fields.

Reply:
@robacarp Interesting, let's run with that. I remember that blog post, actually. And this plus other experience led me to half of this conclusion. Specifically, I've had a bias towards: stick with CRUD, any time you want to add a new action like "pending" then it's a sign you need a new controller. But I had not taken the further step of: even if you are within the CRUD paradigm, if there are multiple views that all #update a model, don't use the same controller for those. Create a new controller for each view.

That's a new consideration, but it's one that feels intuitive. For example, I can imagine the redirect after update being different for these two views. If they're hitting the same endpoint, then you'll end up adding a conditional branch to decide what to do after the update. If they're in separate controllers, then there is a natural separation of concerns.

To state that more broadly: if you found a reason to create two separate views, then it's safe to assume the needs of these views are going to diverge further over time so prepare for that by creating separate controllers.